src/corehaptics.cs

```diff
@@ -0,0 +1,398 @@
+//
+// CoreHaptics C# bindings
+//
+// Authors:
+//	Manuel de la Pena Saenz <mandel@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+
+using System;
+	
+using AVFoundation;
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+namespace CoreHaptics {
+
+// we are not binding the API on Mac OS X yet due to an issue on Apples side: https://github.com/xamarin/maccore/issues/1951
+#if MONOMAC
+	interface AVAudioSession {}
+#endif
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticEventParameter {
+		[BindAs (typeof (CHHapticEventParameterId))]
+		[Export ("parameterID")]
+		NSString ParameterId { get; }
+
```

Comment:
What does the `[BindAs]` attribute do? Does it like map `CHHapticEventParameterId` to `NSString` here? When do we use this?

Reply:
Whenever `NSString` constants are used with enum-like macros or a subclass of `NSString` is used inside ObjC headers.

An API that accept `NSString` is not very useful (e.g. code completion) to _discover_ the right values. For such API we use a _smart_ enum that maps the members with constants. It's _smart_ because the mapping code is generated automatically (so you can't get it wrong by copy/pasting code or forgetting some entries...)